ci/utils.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+is_pod_exist() {
+    local POD_NAME=$1
+    kubectl get pods -A 2>&1 | grep $POD_NAME
+    return $?
+}
+
+is_runtimeclass_exist() {
+    local RUNTIMECLASS_NAME=$1
+    kubectl get runtimeclass 2>&1 | grep $RUNTIMECLASS_NAME
+    returnh $?
+}
+
+wait_pod_ready() {
+    local TIMEOUT_SECS=$1
+    local POD_NAME=$2
+
+    timeout $TIMEOUT_SECS bash -c '
+        pod_name=$0
+        while [ true ]
+        do
+            pod_info=$(kubectl get pods -A 2>&1 | grep $pod_name | awk '"'"'{print $3" "tolower($4)}'"'"')
+            IFS='"'"' '"'"'
+            echo $pod_info
+            read -a temp_arr <<< $pod_info
+            pod_status=${temp_arr[1]}
+            temp=${temp_arr[0]}
+            IFS="/"
+            read -a temp_arr <<< $temp
+            num_pod_ready=${temp_arr[0]}
+            num_pod_total=${temp_arr[1]}
+            
+            echo "[Debug] pod_ready_status: $num_pod_ready/$num_pod_total $pod_status"
+
+            if [[ $num_pod_ready = $num_pod_total && $pod_status = running ]]; then
+                break
+            fi
+
+            sleep 2
+        done
+    ' $POD_NAME
+
+    return $?
+}
+
+wait_runtimeclass_ready() {
+    local TIMEOUT_SECS=$1
+    local RUNTIMECLASS_NAME=$2
+
+    timeout $TIMEOUT_SECS bash -c '
+        runtimeclass_name=$0
+        while [ true ]
+        do
+            kubectl get runtimeclass 2>&1 | grep $runtimeclass_name
```

Comment:
I would prefer `kubectl wait` for all these utils helpers. See e2e.yaml: https://github.com/confidential-containers/operator/blob/6f241fbc056f0a5d9e1bd2c10b2cedc0782b99ff/.github/workflows/e2e.yaml#L51

Reply:
`kubectl get xxx | grep xxxx` can print status of pods as a non-block method, so we can read changing status from Github Actions directly, and analyze which process spends abnormal time. `kubectl wait` only concerns results.

Reply:
> `kubectl wait` only concerns results.

if the expected condition does not trigger you can print out errors and check detailed logs

Reply:
OK.